EduPerfTests/MemoryUsage.cs

```diff
@@ -0,0 +1,144 @@
+
+                            // take a snapshot
+                            b2 = new BrowserMemory(processName);
+
+                            // clear cache and cookies
+                            ClearCookiesAndCache(driver);
+                        }
+
+                        // do work with the memory snapshots
+                        var startSet = b.CurrentPrivateWorkingSet / 1024;
+                        var afterSet = b2.CurrentPrivateWorkingSet / 1024;
+                        var deltaSet = afterSet - startSet;
+                        memData.Iteration = i;
+                        memData.StartMemory = startSet; // default is KB
+                        memData.EndMemory = afterSet;
+                        store.Add(memData);
+                    }
+
+                    // have 5 iterations in memData to get average
+                    long startTimes = 0, endTimes = 0;
+                    foreach (var item in store)
+                    {
+                        startTimes += item.StartMemory;
+                        endTimes += item.EndMemory;
+                    }
+
+                    long startAverage = 0, endAverage = 0;
+                    startAverage = startTimes / iterations;
+                    endAverage = endTimes / iterations;
+                    long delta = endAverage - startAverage;
+
+                    _perfLog.WriteToLog(
+                        site + "," + 
+                        startAverage.ToString() + "," + 
+                        endAverage.ToString() + "," + 
+                        delta.ToString());
+                }
+            }
+
+        }
+
+        private string ProcessNameFromBrowser()
+        {
+            string processName = string.Empty;
+
+            switch (_browser)
+            {
+                case Browser.MicrosoftEdge:
+                    processName = "MicrosoftEdge";
+                    break;
+                case Browser.Chrome:
+                    processName = "chrome.exe";
+                    break;
+                case Browser.Firefox:
+                    processName = "firefox.exe";
+                    break;
+                default:
+                    processName = "MicrosoftEdge";
+                    break;
+            }
+
+            return processName;
+        }
+        private static long LoadEvent(RemoteWebDriver driver)
+        {
+            // This loop is required because Microsoft Edge and Firefox both sometimes return from .Url earlier than they should
+            var loadEventObject = driver.ExecuteScript("return performance.timing.loadEventEnd;");
+            return Convert.ToInt64(loadEventObject);
+        }
+        private static void ClearCookiesAndCache(RemoteWebDriver driver)
+        {
+            try
+            {
+                driver.Manage().Cookies.DeleteAllCookies();
+            }
+            catch (Exception)
+            {
+                // TODO
```

Comment:
It works - most of the time :-) It almost never fails, so I'll write it to console for now...